extensions/GitExtension/FileExplorerGitIntegration/Models/GitLocalRepository.cs

```diff
@@ -68,62 +68,72 @@ IPropertySet ILocalRepository.GetProperties(string[] properties, string relative
             return result;
         }
 
-        foreach (var propName in properties)
+        try
+        {
+            foreach (var propName in properties)
+            {
+                switch (propName)
+                {
+                    case "System.VersionControl.LastChangeMessage":
+                        latestCommit ??= FindLatestCommit(relativePath, repository);
+                        if (latestCommit is not null)
+                        {
+                            result.Add(propName, latestCommit.MessageShort);
+                        }
+
+                        break;
+                    case "System.VersionControl.LastChangeAuthorName":
+                        latestCommit ??= FindLatestCommit(relativePath, repository);
+                        if (latestCommit is not null)
+                        {
+                            result.Add(propName, latestCommit.AuthorName);
+                        }
+
+                        break;
+                    case "System.VersionControl.LastChangeDate":
+                        latestCommit ??= FindLatestCommit(relativePath, repository);
+                        if (latestCommit is not null)
+                        {
+                            result.Add(propName, latestCommit.AuthorWhen);
+                        }
+
+                        break;
+                    case "System.VersionControl.LastChangeAuthorEmail":
+                        latestCommit ??= FindLatestCommit(relativePath, repository);
+                        if (latestCommit is not null)
+                        {
+                            result.Add(propName, latestCommit.AuthorEmail);
+                        }
+
+                        break;
+                    case "System.VersionControl.LastChangeID":
+                        latestCommit ??= FindLatestCommit(relativePath, repository);

```

Comment:
If there's no latest commit (because we're looking at an untracked or ignored file) I'd rather not add the entry to the result at all. But I think this solution cleans it up a bit.